src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticQuerySource.cs

```diff
@@ -78,10 +81,17 @@ public async Task<IQueryResults> ExecuteQueryAsync(Query query, IDictionary<stri
         {
             var results = new List<JsonObject>();
 
-            foreach (var document in docs.TopDocs)
+            foreach (var document in docs.Hits)
             {
-                results.Add(new JsonObject(document.Select(x =>
-                    KeyValuePair.Create(x.Key, (JsonNode)JsonValue.Create(x.Value.ToString())))));
+                var keyValuePairs = document.Source.Select(x =>
+                    KeyValuePair.Create(x.Key, (JsonNode)JsonValue.Create(x.Value.ToString()))).ToList();
```

Comment:
```suggestion
                    KeyValuePair.Create(x.Key, (JsonNode)JsonValue.Create(x.Value.ToString())));
```

Reply:
I need ToList() here to further add two more KeyValuePairs to it with _score and highlight
![image](https://github.com/user-attachments/assets/99e1f4f1-c810-457a-b2e8-36463fbe97c1)